File: EvolutionaryGenerator_v2.1_Python/metah/gp/gp_generator.py
"""Translation of ``mx.tec.meta.gp.GPGenerator``."""

from javacompat import Random
from metah.generator import Generator
from metah.gp.gp_individual import GPIndividual
from metah.individual import Individual


class GPGenerator(Generator):
    """Defines the methods to generate individuals within genetic programing.

    :author: Paola Azeneth Castillo Gutiérrez (Python translation)
    :original author: Jose Carlos Ortiz Bayliss (jcobayliss@tec.mx)
    :original version: 1.0
    """

    MAXDEPTH = 3

    def __init__(self, seed: int):
        """Creates a new instance of ``GPGenerator``.

        :param seed: The seed to initialize the random number generator.
        """
        self.random = Random(seed)

    def generate(self) -> Individual:
        return GPIndividual(GPGenerator.MAXDEPTH, self.random.next_long())

File: EvolutionaryGenerator_v2.1_Java/MetaH/src/mx/tec/meta/gp/GPGenerator.java
package mx.tec.meta.gp;

import java.util.Random;
import mx.tec.meta.Generator;
import mx.tec.meta.Individual;

/**
 * Defines the methods to generate individuals within genetic programing.
 * 
 * @author Jose Carlos Ortiz Bayliss (jcobayliss@tec.mx)
 * @version 1.0
 */
public class GPGenerator extends Generator {

    private final static int MAXDEPTH = 3;
    private final Random random;

    /**
     * Creates a new instance of <code>GPGenerator</code>.
     *
     * @param seed The seed to initialize the random number generator.
     */
    public GPGenerator(long seed) {
        random = new Random(seed);
    }
    
    @Override
    public Individual generate() {
        return new GPIndividual(MAXDEPTH, random.nextLong());
    }
    
}
